packages/editor/src/utils/media-upload/media-upload.js

```diff
@@ -78,15 +78,29 @@ export function mediaUpload( {
 	};
 
 	files.forEach( ( mediaFile, idx ) => {
-		if ( ! isAllowedType( mediaFile.type ) ) {
-			return;
-		}
-
 		// verify if user is allowed to upload this mime type
 		if ( allowedMimeTypesForUser && ! isAllowedMimeTypeForUser( mediaFile.type ) ) {
 			onError( {
 				code: 'MIME_TYPE_NOT_ALLOWED_FOR_USER',
-				message: __( 'Sorry, this file type is not permitted for security reasons.' ),
+				message: [
+					<strong key="filename">{ mediaFile.name }</strong>,
+					': ',
+					__( 'Sorry, this file type is not permitted for security reasons.' ),
+				],
+				file: mediaFile,
+			} );
+			return;
+		}
+
+		// Check if the block supports this mime type
+		if ( ! isAllowedType( mediaFile.type ) ) {
+			onError( {
+				code: 'MIME_TYPE_NOT_SUPPORTED_FOR_BLOCK',
+				message: [
+					<strong key="filename">{ mediaFile.name }</strong>,
+					': ',
+					__( 'Sorry, this file type is not supported by this block.' ),
```

Comment:
We're outside of the block's code in this function, it can be used in a block agnostic context. Should remove "block" from the message and the code?

Reply:
👍🏻